bpf/bpf_lxc.c

```diff
@@ -1436,22 +1437,11 @@ ipv6_policy(struct __ctx_buff *ctx, int ifindex, __u32 src_label,
 skip_policy_enforcement:
 #ifdef ENABLE_NODEPORT
 	if (ret == CT_NEW || ret == CT_REOPENED) {
-		bool dsr = false;
 # ifdef ENABLE_DSR
-		int ret2;
-
-		if (!revalidate_data(ctx, &data, &data_end, &ip6))
-			return DROP_INVALID;
-
-		ret2 = handle_dsr_v6(ctx, ip6, &dsr);
-		if (ret2 != 0)
-			return ret2;
-
-		ct_state_new.dsr = dsr;
-		if (ret == CT_REOPENED && ct_state->dsr != dsr)
-			ct_update_dsr(get_ct_map6(tuple), tuple, dsr);
+		if (ret == CT_REOPENED && ct_state->dsr)
+			ct_update_dsr(get_ct_map6(tuple), tuple, false);
 # endif /* ENABLE_DSR */
-		if (!dsr) {
```

Comment:
Why is the below NodePort logic always executed now when it was only conditionally executed before? What's the consequence of running it even for DSR traffic now?

Reply:
In old code we were able to say "if it's DSR, it's certainly not local-backend".

Now that we no longer look at the connection's DSR state (note that we only flush out any stale `.dsr` flags in the pod-level CT entry, but don't call `ct_has_dsr_egress_entry4()`), *all* connections are potentially for a local-backend.

If this is traffic for a new DSR connection, then the pod-level CT entry has its stale `.node_port` flag cleared.

Reply:
Thanks for the explanation!